bika/lims/browser/client/views/analysisspecs.py

```diff
@@ -5,77 +5,21 @@
 # Copyright 2018 by it's authors.
 # Some rights reserved. See LICENSE.rst, CONTRIBUTORS.rst.
 
+from bika.lims import api
 from bika.lims import bikaMessageFactory as _
-from bika.lims.browser import BrowserView
-from bika.lims.browser.bika_listing import BikaListingView
-from bika.lims.permissions import *
-from bika.lims.utils import isActive
-from bika.lims.utils import tmpID
-from plone.app.layout.globals.interfaces import IViewView
-from Products.CMFCore.utils import getToolByName
-from Products.CMFPlone.utils import _createObjectByType
-from zope.interface import implements
+from bika.lims.controlpanel.bika_analysisspecs import BaseAnalysisSpecsView
+from bika.lims.permissions import AddAnalysisSpec
 
 
-class ClientAnalysisSpecsView(BikaListingView):
-    implements(IViewView)
+class ClientAnalysisSpecsView(BaseAnalysisSpecsView):
 
     def __init__(self, context, request):
         super(ClientAnalysisSpecsView, self).__init__(context, request)
-        self.catalog = 'bika_setup_catalog'
-        self.contentFilter = {
-            'portal_type': 'AnalysisSpec',
-            'sort_on': 'sortable_title',
-            'getClientUID': context.UID(),
-            'path': {
-                "query": "/".join(context.getPhysicalPath()),
-                "level": 0
-            }
-        }
```

Comment:
Add an explicit query to this view. At the moment the path query comes implicitly through the listing view.

Reply:
The base query is fine, I think there is only the need to add "getClientUID" as an additional param here

Reply:
Either this or the physical path. At the moment this works because of this:
https://github.com/senaite/senaite.core/blob/master/bika/lims/browser/bika_listing.py#L520

Reply:
I see, but note `ClientAnalysisSpecsView` inherits from `AnalysisSpecsView` (not directly from `BikaListingView`), so the whole `contentFilter` (with the `path`, `portal_type`, etc.) is already explicited there, so I still don't see why I'd need to copy that again here.